### PR TITLE
feat(add date filter): add calender to andle date filter

### PR DESCRIPTION
**Pull request by oualidelh**
#27

### CODE REVIEW

File: src/components/eventsPageComponents/DateFilter.jsx
Comment (HachemBouhadede) [inline src/components/eventsPageComponents/DateFilter.jsx:1]: we will need a reset date filter button of some kind, once i selected a date i couldn't go back to see all events unless i refresh the page

File: src/components/eventsPageComponents/DateFilter.jsx [hunk 1]
```diff
@@ -1,7 +1,96 @@
...
+    const [today, setToday] = useState(currentDate);
+    const [selectDate, setSelectDate] = useState(currentDate);
+
+    return (
+        <div className='flex justify-center items-center my-4'>
+            <div className='w-[90%] h-auto border rounded-lg  shadow-md'>
+                <div className='flex py-3 md:justify-between justify-center items-center px-6 rounded-t-lg bg-amber-500'>
+                    <h1 className='hidden md:flex select-none font-semibold w-2/3'>
+                        {months[today.month()]} {today.year()}
+                    </h1>
```
Comment (HachemBouhadede) [inline src/components/eventsPageComponents/DateFilter.jsx:20]: i'd suggest we change the location of months and today while keeping the prev and next arrow in other words
Today        <  mm/yy  >

image

File: src/components/eventsPageComponents/EventList.jsx [hunk 2]
```diff
@@ -1,30 +1,77 @@
...
+    const router = useRouter();
+    // sets the routing according to the user's state
+    const handleUser = () => {
+        user
+            ? router.push("/createEvent")
+            : router.push("/authentication/signUp");
+    };
+    // var to handle translatons
+    const { t } = useTranslation();
+    const onClick = (selectedDate) => {
```
Comment (HachemBouhadede) [inline src/components/eventsPageComponents/EventList.jsx:25]: it's better to give functions meaningfull name, i.e handleDateFiltering

File: src/components/eventsPageComponents/EventList.jsx [hunk 3]
```diff
@@ -1,30 +1,77 @@
...
+        const formattedDate = selectedDate.format("DD MMM YYYY");
+        const filteredEvents = events.filter((event) => {
+            return event.date === formattedDate.toLowerCase();
+        });
+        setFilteredEvents(filteredEvents);
+    };
+    // fetch data from dataBase (fireStore)
```
Comment (HachemBouhadede) [inline src/components/eventsPageComponents/EventList.jsx:32]: great logic, simple and straight to the point ~this way the filteredeventslist only displays events of that day tho

File: src/components/util/calendar.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        arrayOfDate.push({
+            currentMonth: true,
+            date: firstDateOfMonth.date(i),
+            today:
+                firstDateOfMonth.date(i).toDate().toDateString() ===
+                dayjs().toDate().toDateString(),
+        });
+    }
+
+    const remaining = 42 - arrayOfDate.length;
```
Comment (HachemBouhadede) [inline src/components/util/calendar.js:33]: not sure about what's this the meaning of this line

**Comment by HachemBouhadede** [comment]
additionaly you have conflicts with the main branch oualid in the translation json files

**Review by Khalil-NOUI — COMMENTED** [review]
nice !

File: src/components/eventsPageComponents/DateFilter.jsx (hunk 1 above)
Comment (oualidelh) [reply src/components/eventsPageComponents/DateFilter.jsx:20]: okay i will work on that

File: src/components/eventsPageComponents/EventList.jsx (hunk 2 above)
Comment (oualidelh) [reply src/components/eventsPageComponents/EventList.jsx:25]: sorry for the confusion but this is for the translation variable

File: src/components/eventsPageComponents/EventList.jsx (hunk 3 above)
Comment (oualidelh) [reply src/components/eventsPageComponents/EventList.jsx:32]: _thanks mate _

File: src/components/util/calendar.js (hunk 4 above)
Comment (HachemBouhadede) [reply src/components/util/calendar.js:33]: this has been explained

File: src/components/eventsPageComponents/DateFilter.jsx (hunk 1 above)
Comment (HachemBouhadede) [reply src/components/eventsPageComponents/DateFilter.jsx]: it has been taken care of !

File: public/locales/ar/common.json
Comment (HachemBouhadede) [inline public/locales/ar/common.json:1]: <3

**Review by HachemBouhadede — APPROVED** [review]

**Comment by HachemBouhadede** [comment]
feel free tomerge the changes, if not merge will be taken care during standup 26/10/2023 10pm

File: src/components/util/calendar.js (hunk 4 above)
Comment (oualidelh) [reply src/components/util/calendar.js:33]: this line is for generate sufix for the calendar in other wrds this generates the days that are not included at this month but in the next month

**Pull request merged**